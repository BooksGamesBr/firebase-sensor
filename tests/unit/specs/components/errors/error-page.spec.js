import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import ErrorPage from '@/components/errors/error-page'
import { FakeErrors } from '../../../data/fake-data'

function genlocalVue () {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.filter('formatDate', (value) => {
    if (value) {
      return new Date(value * 1000).toLocaleString('en-US')
    }
  })
  return localVue
}

describe('ErrorPage.vue', () => {
  let getters
  let actions
  let store
  let localVue

  beforeEach(() => {
    localVue = genlocalVue()
    getters = {
      errors: () => FakeErrors
    }
    actions = {
      removeError: jest.fn()
    }

    store = new Vuex.Store({
      getters,
      actions
    })
  })

  it('shows a page', () => {
    const wrapper = shallowMount(ErrorPage, {
      store,
      localVue
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should call the remove action when the bin icon is clicked on', () => {
    const wrapper = shallowMount(ErrorPage, {
      store,
      localVue
    })
    const deleteIcon = wrapper.find('.col-lg-12:first-child .glyphicon-trash')
    deleteIcon.trigger('click')

    expect(actions.removeError).toHaveBeenCalled()
  })
})
