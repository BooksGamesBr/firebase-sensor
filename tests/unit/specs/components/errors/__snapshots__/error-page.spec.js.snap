// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage.vue shows a page 1`] = `
<div class="row">
  <div class="col-lg-12">
    <div role="alert" class="alert alert-danger"><span class="glyphicon glyphicon-trash" style="cursor: pointer;"></span>
      <b>2018-2-13 11:47:51 / </b>Error 1
    </div>
  </div>
  <div class="col-lg-12">
    <div role="alert" class="alert alert-danger"><span class="glyphicon glyphicon-trash" style="cursor: pointer;"></span>
      <b>2018-2-13 11:48:51 / </b>Error 2
    </div>
  </div>
</div>
`;
